### PR TITLE
case-insensitive origin compare

### PR DESCRIPTION
**Pull request by mdgbayly**
Modified port.validateEvent so that it does case insensitive comparisons on targetOrigin and event.origin.

Added unit tests including tests for null/undefined. Not sure if this is overkill?

Also modified the test:unit command so that it runs on windows.
Not sure if this will break the Mac tests?? So you can ignore if it will.

### CODE REVIEW

File: src/port.js [hunk 1]
```diff
@@ -265,7 +265,7 @@ export default class Port {
 	}
 	static validateEvent(targetOrigin, endpoint, e) {
 		var isValid = (e.source === endpoint) &&
-			(targetOrigin === '*' || targetOrigin === e.origin) &&
+			(targetOrigin === '*' || targetOrigin != null && e.origin != null && targetOrigin.toUpperCase() === e.origin.toUpperCase()) &&
```
Comment (dlockhart) [inline src/port.js]: You'll want a typesafe comparison here: `targetOrigin !== null && e.origin !== null`

File: package.json [hunk 2]
```diff
@@ -11,7 +11,7 @@
     "lint:test": "jshint test/*.js",
     "prepublish:cdn": "npm run browserify",
     "publish:cdn": "gulp publish",
-    "test:unit": "istanbul cover _mocha -- --recursive",
+    "test:unit": "istanbul cover node_modules/mocha/bin/_mocha -- --recursive",
```
Comment (dlockhart) [inline package.json]: Unfortunate that Windows requires this... but seems to still work elsewhere so seems fine.

**Comment by dlockhart** [comment]
Also, feel free for this to be the one place you don't need to worry about Rally user story numbers in branch names or commits. :)

File: src/port.js (hunk 1 above)
Comment (dlockhart) [inline src/port.js]: I'm guessing once you add that, your `undefined` cases will start to fail, so you'll need to either explicitly check for `undefined` first in addition to `null`, or:

    ( targetOrigin === '*' || targetOrigin && targetOrigin.toUpperCase() === e.origin.toUpperCase() )

That might simplify things a bit (if it works and passes the tests). It should handle `undefined`, `null` and empty string.

**Comment by mdgbayly** [comment]
That was actually still a work in progress - I didn't realized re-pushing my branch would update the pull request.

I was surprised I had to add the === true, but without it the unit tests were failing because the long conditional was setting isValid to null or undefined if targetOrigin or e.origin were null or undefined.

e.g. 
var isValid = ("a" === "a" && undefined) => equals undefined not false
var isValid = ("a" === "a" && null) => equals null not false

The alternative was to explicitly test for !== null && !== undefined but that didn't cover the empty string use case. I was looking into a cleaner solution to use an isStringEmpty function instead which I'll push now.


The alternative was to switch from

**Comment by mdgbayly** [comment]
Ah ok - just saw your message about skipping Rally user stories! Ok :-)

**Comment by dlockhart** [comment]
Cool looks good, let me know if you're ready and I'll merge this in!
